controllers/hcpvaultsecretsapp_controller.go

```diff
@@ -123,6 +129,39 @@ func (r *HCPVaultSecretsAppReconciler) Reconcile(ctx context.Context, req ctrl.R
 		return ctrl.Result{}, r.handleDeletion(ctx, o)
 	}
 
+	// keep track of the HCPVaultSecretsApp for the cleanup goroutine
+	if HCPVaultSecretsAppMap[req.NamespacedName.String()] == nil {
```

Comment:
I create a map here instead of using the existing `HCPVaultSecretsAppList` because it made it easier to keep track of any HVS apps (I would have needed a loop here instead of simply an if statement).